### ISSUE TITLE
sysctl module should record not compliant when unknown key found

### ISSUE THREAD

**Issue by dgmorrisjr** [HubbleStack/Nova#58]
On a system that doesn't have a sysctl key defined, hubble.nova doesn't provide any feedback.  

here is the output from the hubble.nova command:

```
[root@master ~]# salt master hubble.audit tag=*4.2.4
master:
    ----------
    Failure:
    Success:
```

Here is the output of running the `/sbin/sysctl` command directly, indicating that the key does not exist in the config file.  

```
[root@master ~]# /sbin/sysctl net.ipv4.conf.all.log_martian
error: "net.ipv4.conf.all.log_martian" is an unknown key
```

Further testing has shown that including entries for which there are no tags to check such as the 4.2.4 above, will cause the entire run to not return anything.  Ideally, the module should still indicate the control number (e.g. CIS-4.2.4) and the commands return status (e.g. unknown key).

**Comment by gmiu** [comment]
@dgmorrisjr, is the grain value for `osfinger` set correctly in the yaml file you're using here?

**Comment by dgmorrisjr** [comment]
@avb76, I believe so, I've looked over this multiple times and tried many variations... certainly possible i'm overlooking/missing something.

Here is the block from the yaml file.  When this block is commented out, everything works fine:

```
log_suspicious_packets:
  data:
    'Red Hat Enterprise Linux Server-6':
      - 'net.ipv4.conf.all.log_martian':
          tag: 'CIS-4.2.4'
          match_output: '1'
      - 'net.ipv4.conf.default.log_martians':
          tag: 'CIS-4.2.4'
          match_output: '1'
    'CentOS-6':
      - 'net.ipv4.conf.all.log_martian':
          tag: 'CIS-4.2.4'
          match_output: '1'
      - 'net.ipv4.conf.default.log_martians':
          tag: 'CIS-4.2.4'
          match_output: '1'
    'CentOS Linux-7':
      - 'net.ipv4.conf.all.log_martian':
          tag: 'CIS-4.2.4'
          match_output: '1'
      - 'net.ipv4.conf.default.log_martians':
          tag: 'CIS-4.2.4'
          match_output: '1'
  description: 'Log Suspicious Activity (Scored)'
  alert: email
  trigger: state
```

**Comment by gmiu** [comment]
Yeah, it should be `net.ipv4.conf.all.log_martians` instead of `net.ipv4.conf.all.log_martian` in the yaml file. I'll correct that now. But you should still get a failure on this rule. I'll do some more tests on this one.

**Comment by cedwards** [comment]
A quick check shows that `log_martian` isn't valid, but `log_martians` (plural) is valid. I'm not sure if that's a typo in the PDF or in the translation to the YAML. In any case we need to better capture unknown key errors and @avb76 will take care of it.

(@dgmorrisjr if you're interested we have a Slack team for HubbleStack which may make some of this back and forth easier. Check out the footer at https://hubblestack.io)

**Comment by dgmorrisjr** [comment]
I fixed the typeo and that one seems to work now.  I had copied the block from the CentOS Linux-7 block.

**Comment by gmiu** [comment]
I found another bug while trying to fix this one: if one of those two sysctl checks fails and the other one succeeds, you'll get the same tag for both `Failure` and `Success` lists. Fixing this one as we speak.

**Comment by gmiu** [comment]
Fixed in #62
